api/src/pcapi/routes/backoffice/chronicles/blueprint.py

```diff
@@ -147,5 +228,91 @@ def unpublish_chronicle(chronicle_id: int) -> utils.BackofficeResponse:
         author=current_user,
         chronicle=chronicle,
     )
-    flash(f"La chronique {chronicle_id} à été dépubliée", "success")
-    return redirect(url_for("backoffice_web.chronicles.list_chronicles"), code=303)
+    flash(f"La chronique {chronicle_id} a été dépubliée", "success")
+    return redirect(request.referrer or url_for("backoffice_web.chronicles.list_chronicles"), code=303)
+
+
+@chronicles_blueprint.route("/<int:chronicle_id>/attach-product", methods=["POST"])
+@atomic()
+@permission_required(perm_models.Permissions.MANAGE_CHRONICLE)
+def attach_product(chronicle_id: int) -> utils.BackofficeResponse:
+    if not FeatureToggle.WIP_ENABLE_CHRONICLES_IN_BO.is_active():
+        raise NotFound()
+
+    form = forms.AttachProductForm()
+    if not form.validate():
+        mark_transaction_as_invalid()
+        flash(utils.build_form_error_msg(form), "warning")
+        redirect(url_for("backoffice_web.chronicles.details", chronicle_id=chronicle_id), code=303)
+
+    chronicle = chronicles_models.Chronicle.query.get_or_404(chronicle_id)
+    products = offers_models.Product.query.filter(offers_models.Product.extraData["ean"].astext == form.ean.data).all()
+
+    if not products:
+        mark_transaction_as_invalid()
+        flash("Aucune œuvre n'a été trouvée pour cet EAN", "warning")
+        return redirect(
+            url_for("backoffice_web.chronicles.details", chronicle_id=chronicle_id, active_tab="book"), code=303
+        )
+
+    for product in products:
+        chronicle.products.append(product)
+        db.session.add(product)
+        flash(
+            Markup("Le produit <b>{product_name}</b> a été rattaché à la chronique").format(product_name=product.name),
+            "success",
+        )
```

Comment:
Je pense que pour éviter de multiplier les messages flash, on aurait pu faire plus simplement, avec quelque chose du style (non testé) : 
```suggestion
    chronicle.products += products
    db.session.add_all(products)
    flash(
        Markup(
            "Les produits <b>{product_names}</b> ont été rattachés à la chronique").format(product_names=", ".join(product.name for product in producs))
            if len(products) > 1
            else Markup("Le produit <b>{product_name}</b> a été rattaché à la chronique").format(product_name=product.name),
            "success",
    )
```